lib/vagrant-libvirt/action/prepare_nfs_settings.rb

```diff
@@ -59,7 +59,7 @@ def read_machine_ip(machine)
           return ssh_host if ping(ssh_host)
 
           # check other ips
-          command = "ip addr show | grep -i 'inet ' | grep -v '127.0.0.1' | tr -s ' ' | cut -d' ' -f3 | cut -d'/' -f 1"
+          command = "/sbin/ip addr show | grep -i 'inet ' | grep -v '127.0.0.1' | tr -s ' ' | cut -d' ' -f3 | cut -d'/' -f 1"
```

Comment:
Wouldn't that suffer the same problem? If `ip` is not in the path `which` won't return anything?

Perhaps you meant `ip=$(which ip); ${ip:-/sbin/ip} addr show | ....` so that it would only default to attempting to use `/sbin/ip` if it couldn't find `ip` in the path.